options/rtdl/generic/linker.hpp

```diff
@@ -77,10 +77,10 @@ enum class HashStyle {
 using InitFuncPtr = void (*)();
 
 struct SharedObject {
-	SharedObject(const char *name, bool is_main_object,
+	SharedObject(frg::string<MemoryAllocator> name, bool is_main_object,
```

Comment:
Maybe you could have a second constructor taking a ``const char *`` to clear up the ``frg::string`` instantiations you currently do across the code